### PR TITLE
V5.2.1 static gklib

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] ~Bumped the build number (if the version is unchanged)~
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closs #33, closes #41

### CODE REVIEW

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
This is nearing completion. @matthiasdiener, @minrk, since this is incorporating a lot of your work from #41, #33, and the gklib-feedstock, would you like to become maintainers here?

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
Currently, this adds @matthiasdiener as maintainer, but not @minrk. This is taken from the last state in #41 and can of course be adapted, but I think it should not stop us from merging if happy otherwise; any further change can be handled in a separate PR.

**Comment by minrk** [comment]
I don't need to be a maintainer. Thanks for asking! In general, conda-forge discourages static libs, so if gklib works as a shared library, that seems preferable. But if it doesn't really, and it's always meant to be used statically in practice, then I suppose this is okay. I'll leave it to metis/gklib folks to decide.

But if it's mostly a matter of missing patches applied in in the bundled gklib build here and not in https://github.com/conda-forge/gklib-feedstock, I'd argue that the fix is to submit the necessary patches to gklib-feedstock (and upstream to gklib itself, if appropriate).

**Comment by zklaus** [comment]
We could integrate the patches into the gklib feedstock. The issue is that upstream hasn't cut a release in a long time. That leaves us in an uncomfortable position because we can now either publish as a `post` build of the old version with quite a bit of patching, or try to make the old version work again with minimal patching (a lot of work). Plus I think we determined that Metis really is the only user of GKlib and this vendoring is what's been done up until now anyways.

**Comment by minrk** [comment]
Yeah, if it's mostly a local factorization of one project and not really separate (especially if it doesn't really work as a shared library), then I'm happy with it being a local statically linked dependency.

**Review by minrk — APPROVED** [review]
I'm happy with this, I'll let other maintainers finish reviewing and merge.

We can always link gklib dynamically in a future release if/when upstream matures the separation of the two projects.

File: recipe/meta.yaml
Comment (jaimergp) [inline recipe/meta.yaml:72]: I think we need to revert this change given how we are back to static.

**Pull request merged**
